Classes/FileReference.php

```diff
@@ -37,18 +37,18 @@ public function __construct(array $fileReferenceData, $factory = null)
 
 		$file = Environment::getPublicPath() . '/' . $this->publicUrl;
 
-		if (!file_exists($file)) {
-			if (isset($GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['loadFilesFromRemoteIfMissing']) &&
-				$GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['loadFilesFromRemoteIfMissing']) {
-				$this->loadFileFromRemote();
-			}
+		if (file_exists($file)) {
+			return;
 		}
 
-		if (!file_exists($file)) {
-			if (isset($GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['fakeFilesIfMissing']) &&
-				$GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['fakeFilesIfMissing']) {
-				$this->createFakeFile();
-			}
+		if (isset($GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['loadFilesFromRemoteIfMissing']) &&
+			$GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['loadFilesFromRemoteIfMissing']) {
+			$this->loadFileFromRemote();
+		}
+
+		if (isset($GLOBALS['TYPO3_CONF_VARS']['SYS']['fal']['fakeFilesIfMissing']) &&
```

Comment:
@AMartinNo1, be careful ;-)

With this change and having both options activated a non-existing file will be fetched from the remote server and then immediately be overwritten by the fake file...

Reply:
Good point!